frontend/lib/src/components/transaction_component/transaction_component.dart

```diff
@@ -6,56 +9,109 @@ import 'package:get/get.dart';
 
 /*
   Reusable transaction component for deposit, withdraw, and transfer cash
 
   @param label: String value to display what kind of transaction it is,
     basically something like a title.
 
-  @param type: String value representing the type of transaction: DEPOSIT, WITHDRAWAL, or TRANSFER.
+  @param type: String value representing the type of transaction: DEPT, CREDIT, or TRANSFER.
     Determines the fields to display and the validation logic.
+    Value should follow the backend enum values.
+
+  @param accountId: String value for the id of the user's account
 */
 
 class TransactionComponent extends StatelessWidget {
   const TransactionComponent({
     Key? key,
     required this.label,
-    this.type = 'DEPOSIT',
+    this.type = 'DEPT',
+    required this.accountId,
   }) : super(key: key);
 
   final String label;
   final String type;
+  final String accountId;
 
   @override
   Widget build(BuildContext context) {
     final formKey = GlobalKey<FormBuilderState>();
+    final DashboardController controller = Get.put(DashboardController());
+
+    void alertDialog({String? title, String? content}) => showAlertDialog(
+          title: title ?? 'Invalid Input',
+          content: content ?? 'Please make sure to fill out the field',
+        );
+
+    void onSubmit() async {
+      await NetworkConfig().getCsrftoken();
+      final client = NetworkConfig().client;
 
-    void onSubmit() {
       if (formKey.currentState?.isValid ?? false) {
-        final emailValue = formKey.currentState?.fields['email']?.value;
+        final receiverValue =
+            formKey.currentState?.fields['receiver_id']?.value;
         final amountValue = formKey.currentState?.fields['amount']?.value;
+        final descriptionValue =
+            formKey.currentState?.fields['description']?.value;
+        // Manually set category for now as there is no plan for this feature as of now
+        // TODO: Deposit and transfer
+        final Map typeTexts = {
+          'CREDIT': {'category': 'BILLS', 'success': 'withdrawn'},
+        };
 
-        // if field is null, do not execute Get.back();
-        if ((type == 'TRANSFER' &&
-                (emailValue == null || amountValue == null)) ||
-            (type == 'DEPOSIT' && amountValue == null)) {
-          showAlertDialog(
-            title: 'Invalid Input',
-            content: 'Please make sure to fill out the field',
-          );
+        // If field is null, do not execute Get.back();
+        if (amountValue == null ||
+            descriptionValue == null ||
+            (type.contains('TRANSFER') && receiverValue == null)) {
+          alertDialog();
 
           return;
         }
 
-        Get.back();
+        final transactionResponse = await client.post(
+          accountTransactionsUrl.replaceFirst('{id}', accountId),
+          data: {
+            'amount': amountValue,
+            'description': descriptionValue,
+            'transaction_type': type,
```

Comment:
for transaction type, please use enum. you can refer to my draft PR for transfer integration, and check the transaction model.

Reply:
Removed